lighthouse.go

```diff
@@ -203,70 +211,47 @@ func (lh *LightHouse) DeleteVpnIP(vpnIP uint32) {
 	lh.Unlock()
 }
 
-// AddRemote is correct way for non LightHouse members to add an address. toAddr will be placed in the learned map
-// static means this is a static host entry from the config file, it should only be used on start up
-func (lh *LightHouse) AddRemote(vpnIP uint32, toAddr *udpAddr, static bool) {
-	if ipv4 := toAddr.IP.To4(); ipv4 != nil {
-		lh.addRemoteV4(vpnIP, NewIp4AndPort(ipv4, uint32(toAddr.Port)), static, true)
-	} else {
-		lh.addRemoteV6(vpnIP, NewIp6AndPort(toAddr.IP, uint32(toAddr.Port)), static, true)
-	}
-
-	//TODO: if we do not add due to a config filter we may end up not having any addresses here
-	if static {
-		lh.staticList[vpnIP] = struct{}{}
-	}
-}
-
-// unlockedGetAddrs assumes you have the lh lock
-func (lh *LightHouse) unlockedGetAddrs(vpnIP uint32) *ip4And6 {
-	am, ok := lh.addrMap[vpnIP]
-	if !ok {
-		am = &ip4And6{}
-		lh.addrMap[vpnIP] = am
-	}
-	return am
-}
-
-// addRemoteV4 is a lighthouse internal method that prepends a remote if it is allowed by the allow list and not duplicated
-func (lh *LightHouse) addRemoteV4(vpnIP uint32, to *Ip4AndPort, static bool, learned bool) {
-	// First we check if the sender thinks this is a static entry
-	// and do nothing if it is not, but should be considered static
-	if static == false {
-		if _, ok := lh.staticList[vpnIP]; ok {
-			return
-		}
-	}
-
+// AddStaticRemote adds a static host entry for vpnIp as ourselves as the owner
+// We are the owner because we don't want a lighthouse server to advertise for static hosts it was configured with
+// And we don't want a lighthouse query reply to interfere with our learned cache if we are a client
+func (lh *LightHouse) AddStaticRemote(vpnIp uint32, toAddr *udpAddr) {
 	lh.Lock()
-	defer lh.Unlock()
-	am := lh.unlockedGetAddrs(vpnIP)
+	am := lh.unlockedGetRemoteList(vpnIp)
+	am.Lock()
+	defer am.Unlock()
+	lh.Unlock()
 
-	if learned {
-		if !lh.unlockedShouldAddV4(am.learnedV4, to) {
+	if ipv4 := toAddr.IP.To4(); ipv4 != nil {
+		to := NewIp4AndPort(ipv4, uint32(toAddr.Port))
+		if !lh.unlockedShouldAddV4(to) {
 			return
 		}
-		am.learnedV4 = prependAndLimitV4(am.learnedV4, to)
+		am.unlockedPrependV4(lh.myVpnIp, to)
+
 	} else {
-		if !lh.unlockedShouldAddV4(am.v4, to) {
+		to := NewIp6AndPort(toAddr.IP, uint32(toAddr.Port))
+		if !lh.unlockedShouldAddV6(to) {
 			return
 		}
-		am.v4 = prependAndLimitV4(am.v4, to)
+		am.unlockedPrependV6(lh.myVpnIp, to)
 	}
+
+	// Mark it as static
+	lh.staticList[vpnIp] = struct{}{}
```

Comment:
we don't have the lh lock when we call this, I think we probably want it.

Reply:
It's a bit of a hand wave, the only time this function is called is on boot pre-start (currently)

Reply:
The better way to deal with this is to move this into RemoteList as a bool